src/marqo/tensor_search/tensor_search.py

```diff
@@ -931,7 +934,7 @@ def _get_documents_for_upsert(
         for result in result_list:
             if result["found"]:
                 doc_in_results = True
-                if result["_source"]["__chunks"] == []:
+                if ("__chunks" in result["_source"]) and (result["_source"]["__chunks"] == []):
                     res_data = result
                 else:
                     res_chunks = result
```

Comment:
Are we guaranteed that `"__chunks" not in result["_source"]` implies `result` is meant to be tensor request (as opposed to a body request)?

Reply:
This is because the idea is for `res_chunks`, we only include chunk-related content like
```
"_source": {"include": [f"__chunks.__field_content", f"__chunks.__field_name", f"__chunks.__vector_*"]}
```

In our testing, we only have 1 scenario wherein `"__chunks"` is not in the result, and that is when we make a tensor request and no tensors exist. When we do a body request, the behavior is that `"__chunks"` is always there, it's just empty. This is the best way I found to differentiate between a tensor and body request, and it passes all the edge cases I can think of